components/brave_wallet_ui/components/market-datatable/index.tsx

```diff
@@ -0,0 +1,170 @@
+    currentPrice,
+    priceChange24h,
+    priceChangePercentage24h,
+    marketCap,
+    marketCapRank,
+    totalVolume
+  } = coinMarkDataItem
+
+  const formattedPrice = new Amount(currentPrice).formatAsFiat('USD')
+  const formattedPercentageChange = new Amount(priceChangePercentage24h).value?.absoluteValue().toFixed(2) + '%'
+  const formattedMarketCap = new Amount(marketCap).abbreviate(1, 'USD', 'billion')
+  const formattedVolume = new Amount(totalVolume).abbreviate(1, 'USD')
+  const isDown = priceChange24h < 0
+
+  const cellsContent: React.ReactNode[] = [
+    <AssetsColumnWrapper>
+      {/* Hidden until wishlist feature is available on the backend */}
+      {/* <AssetsColumnItemSpacer>
+          <AssetWishlistStar active={true} />
+        </AssetsColumnItemSpacer> */}
+      <AssetsColumnItemSpacer>
+        <TextWrapper alignment="center">{marketCapRank}</TextWrapper>
+      </AssetsColumnItemSpacer>
+      <AssetNameAndIcon
+        assetName={name}
+        symbol={symbol}
+        assetLogo={image}
+      />
+    </AssetsColumnWrapper>,
+
+    // Price Column
+    <TextWrapper alignment="right">{formattedPrice}</TextWrapper>,
+
+    // Price Change Column
+    <TextWrapper alignment="right">
+      <AssetPriceChange
+        isDown={isDown}
+        priceChangePercentage={formattedPercentageChange}
+      />
+    </TextWrapper>,
+
+    // Market Cap Column
+    <TextWrapper alignment="right">{formattedMarketCap}</TextWrapper>,
+
+    // Volume Column
+    <TextWrapper alignment="right">{formattedVolume}</TextWrapper>
+
+    // Line Chart Column
+    // Commented out because priceHistory data is yet to be
+    // available from the backend
+    // <LineChartWrapper>
+    //   <LineChart
+    //     priceData={priceHistory}
+    //     isLoading={false}
+    //     isDisabled={false}
+    //     isDown={isDown}
+    //     isAsset={true}
+    //     onUpdateBalance={() => {}}
+    //     showPulsatingDot={false}
+    //     showTooltip={false}
+    //     customStyle={{
+    //       height: '20px',
+    //       width: '100%',
+    //       marginBottom: '0px'
+    //     }}
+    //   />
+    // </LineChartWrapper>
+  ]
+
+  const cells: Cell[] = cellsContent.map(cellContent => {
+    return {
+      content: cellContent
+    }
+  })
+
+  return cells
+}
+
+export const MarketDataTable = (props: Props) => {
+  const { headers, coinMarketData, moreDataAvailable, showEmptyState, onShowMoreCoins, onSort, onSelectCoinMarket } = props
+  const ref: any = React.useRef<HTMLDivElement>()
+  const onScreen = useOnScreen<HTMLDivElement>(ref, onScreenOptions)
+
+  React.useEffect(() => {
+    if (onScreen && moreDataAvailable) {
+      onShowMoreCoins()
+    }
+  }, [onScreen, moreDataAvailable, onShowMoreCoins])
+
+  const rows: Row[] = React.useMemo(() => {
+    return coinMarketData.map((coinMarketItem: BraveWallet.CoinMarket) => {
+      return {
+        id: `coin-row-${coinMarketItem.symbol}-${coinMarketItem.marketCapRank}`,
```

Comment:
This `id` is not meant for location since in React we don't use direct DOM manipulation. The main purpose for this id is for unique identification of the element during re-renders and it is used my react internals. We are appending `symbol` and `marketCapRank` to ensure the `id` is always unique for each coin.

Reply:
This `id` is not meant for location since in React we don't use direct DOM manipulation. The main purpose for this id is for unique identification of the element during re-renders and it is used my react internals. We are appending `symbol` and `marketCapRank` to ensure the `id` is always unique for each coin.

Reply:
Sorry I didn't mean this in the since of DOM location, I meant this as being an identifier that can be used to locate (in the sense of a URL is a "universal resource locator") the associated data for the identifier later. So for example, If the identifier for ETH is `coin-row-ETH-2` today and then shifts tomorrow to `coin-row-ETH-3` or `coin-row-ETH-1` will that cause an issue where the data it's identified is either unable to be found on lookup or alternatively cause it to redundantly store the state for this in the page such that two ETH tokens show up later?